### ISSUE TITLE
Improving the Visualization and User Interface for Page Images

### ISSUE THREAD

**Issue by stweil** [kitodo/kitodo-presentation#923]
## Description

Displaying page images is one of the main tasks of Kitodo.Presentation and should be as user friendly as possible.

It must support desktop and mobile devices, different screen sizes, user interaction via mouse / touchscreen / touchpad / trackball / keyboard, be responsive for different window sizes and more.

Today it is sometimes annoying. Double page view does not automatically select the correct left and right side, zooming is not smooth, and the page image can "fly" out of the visible window.

Technically Kitodo.Presentation uses OpenLayers for the images presentation. Maybe there exist better libraries that could be offered as an alternative.

## Related Issues

None known.

## Expected Benefits of this Development

The user experience for Kitodo.Presentation should be as good as possible.

## Estimated Costs and Complexity

Depending on the improvements implemented I expect costs to range from medium (less than 10 days) to high (more than 10 days).

**Comment by stweil** [comment]
## Demo

https://user-images.githubusercontent.com/6734573/223210855-845f9174-8c82-4fac-9405-c139738420b8.mov

**Comment by stweil** [comment]
I am afraid that all implementations which use Kitodo.Presentation currently show the problems which I have described above. Maybe we can add examples of better user interfaces from other libraries here as an inspiration for Kitodo.

**Comment by sebastian-meyer** [comment]
I completely agree with the proposal to make the page view more accessible and user friendly, but I don't think this requires changing the underlying framework. OpenLayers already offers all the features we need including support for touchscreens/touchpads, responsive designs, restricting zoom levels according to image resolutions, always keeping the image inside the viewport, etc. In fact, support for touchscreens and responsive designs are already implemented in Kitodo.Presentation - the issue here is that most instances of Kitodo.Presentation don't restrict touch input exclusively to the page view, leading to an awkward experience where for example pinch-zooming not only zooms the page image, but also the surrounding layout resulting in seemingly uncontrolled zooming jumping between different zoom levels.

I suggest keeping OpenLayers, but improving its integration with Kitodo.Presentation in order to address the usability issues.

**Comment by sebastian-meyer** [comment]
Votes: 2

**Comment by stweil** [comment]
Please add the label for development fund 2024.

**Comment by sebastian-meyer** [comment]
Votes: 0

**Comment by stweil** [comment]
I think that a new demo or test implementation like #1176 must address  this issue.
